plugins/inputs/syslog/syslog.go

```diff
@@ -156,21 +156,45 @@ func (s *Syslog) Stop() {
 func (s *Syslog) createStreamDataHandler(acc telegraf.Accumulator) socket.CallbackConnection {
 	// Create parser options
 	var opts []syslog.ParserOption
-	if s.BestEffort {
-		opts = append(opts, syslog.WithBestEffort())
+	var machineOpts []syslog.MachineOption
+	switch s.SyslogStandard {
+	case "RFC3164":
+		if s.BestEffort {
+			machineOpts = append(machineOpts, rfc3164.WithBestEffort())
+		}
+		machineOpts = append(machineOpts, rfc3164.WithYear(rfc3164.CurrentYear{}))
```

Comment:
Consideration: For RFC3164, you're using `rfc3164.CurrentYear{}` - is this the desired behavior? RFC3164 messages don't include year information, so this assumes current year. This seems reasonable but worth documenting.

Reply:
I'm replicating the behaviour in the UDP handler, see line 228 in `createDatagramDataHandler`. Do you think it's possible to factor out the machine options for it to be common and/or configurable? It's not obvious to me right now.